### PR TITLE
[Avatar] Remove customer prop and experiemntal values

### LINKED ISSUE #9977: Avatar - shape to variant, customer to variant (needs design)
```[tasklist]
### Prop changes
- [x] `shape` to `variant` and remove `round` variant or remove shape altogether
- [x] `customer` boolean to `variant` (needs design to see if possible)
- [x] convert `-experimental` sizes to normal size naming convention
```

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes #9977

### WHAT is this pull request doing?

- [x] Remove `customer` boolean prop (make the default)
- [x] Remove `-experimental` from `size` prop
- [x] Update Avatar stories
- [x] Update Avatar documentation examples
- [x] Update `v12` migration guide
- [x] Write migration for `Avatar` prop changes 
  - [x] remove `customer` prop
  - [x] `size` prop mapping

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/hot-apples-press.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': major
+---
+
+Removed experimental suffix from Avatar size values
```
Comment (kyledurand) [inline .changeset/hot-apples-press.md]: Can we also add the size name changes? I think those are more important to document than experimental changes

File: polaris-react/src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -1,19 +1,13 @@
...
 
 import styles from './Avatar.scss';
 
-type Size =
-  | 'extraSmall'
-  | 'small'
-  | 'medium'
-  | 'large'
-  | Experimental<'xl' | '2xl'>;
+type Size = 'xs' | 'small' | 'medium' | 'large' | 'xl' | '2xl';
```
Comment (kyledurand) [inline polaris-react/src/components/Avatar/Avatar.tsx]: How did we determine these size names? These are half verbose half tshirt sizes which seems weird to me. Are we doing this somewhere else? Should we match breakpoint naming conventions?

```suggestion
type Size = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```

**Review by kyledurand — COMMENTED** [review]
This looks good just a bit confused on the naming

File: .changeset/hot-apples-press.md (hunk 1 above)
Comment (sam-b-rose) [reply .changeset/hot-apples-press.md]: Yes, good catch 👍

File: polaris-react/src/components/Avatar/Avatar.tsx (hunk 2 above)
Comment (sam-b-rose) [reply polaris-react/src/components/Avatar/Avatar.tsx]: I was also confused by this. @aaronccasanova or @sarahill might have more context. Aligning with the breakpoints makes sense to me 👍

Comment (sarahill) [reply polaris-react/src/components/Avatar/Avatar.tsx]: I don't have context on why we went with those naming conventions. The suggested update makes sense to me and follows the convention we usually use 👍

Comment (sam-b-rose) [reply polaris-react/src/components/Avatar/Avatar.tsx]: Excellent, agreed! I will revise to match the breakpoints then. Thanks @sarahill @kyledurand

File: polaris-react/src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
 
 import styles from './Avatar.scss';
 
-type Size =
-  | 'extraSmall'
-  | 'small'
-  | 'medium'
-  | 'large'
-  | Experimental<'xl' | '2xl'>;
+type Size = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
```
Comment (kyledurand) [inline polaris-react/src/components/Avatar/Avatar.tsx:10]: Do we not need `2xl` anymore? Looks like there are still mentions of it in the migration table and some other parts of this PR

File: polaris-react/src/components/Avatar/Avatar.stories.tsx [hunk 4]
```diff
@@ -23,12 +23,12 @@ export default {
 const sizes: {
   [S in NonNullable<AvatarProps['size']>]: string;
 } = {
-  '2xl-experimental': 'XXL',
-  'xl-experimental': 'XL',
+  '2xl': 'XXL',
```
Comment (kyledurand) [inline polaris-react/src/components/Avatar/Avatar.stories.tsx]: Can we remove this?

**Review by kyledurand — APPROVED** [review]
Thanks Sam 🙇

File: polaris-react/src/components/Avatar/Avatar.stories.tsx (hunk 4 above)
Comment (sam-b-rose) [reply polaris-react/src/components/Avatar/Avatar.stories.tsx]: Good catch, yes!

**Pull request merged**